ckanext/superset/decorators.py

```diff
@@ -9,10 +9,25 @@ def require_sysadmin_user(func):
 
     @wraps(func)
     def view_wrapper(*args, **kwargs):
-        if not hasattr(toolkit.c, "user") or not toolkit.c.user:
-            return toolkit.abort(403, "Forbidden")
-        if not toolkit.c.userobj.sysadmin:
+        user_name = getattr(toolkit.c, "user", None)
+        print(f"\n🔍 Usuario detectado en CKAN: {user_name}")
+
+        if not user_name:
+            print("⛔ No se detectó un usuario autenticado.")
+            return toolkit.abort(403, "Forbidden: No user detected")
+
+        try:
+            user_data = toolkit.get_action('user_show')({}, {'id': user_name})
```

Comment:
Why is `user_show` required?
Test with `current_user` and if fails, let's discuss it

Reply:
Continue here

Reply:
@avdata99 

With this code `current_user` I get this error

`------------------------------------------------------------------------------------------------ Captured log call ------------------------------------------------------------------------------------------------
ERROR    ckanext.superset.decorators:decorators.py:23 Error getting user data: "Action 'current_user' not found"
INFO     ckan.config.middleware.flask_app:flask_app.py:491 403 Forbidden: Forbidden: User not found
INFO     ckan.config.middleware.flask_app:flask_app.py:401  403 /apache-superset/create-dataset/32 render time 0.053 seconds
`

Reply:
`current_user` is not an action
Look at this CKAN core example using it: https://github.com/ckan/ckan/blob/master/ckan/lib/helpers.py#L2092